doc/design/parallel-node-config.md

```diff
@@ -0,0 +1,216 @@
+---
+title: Parallel SR-IOV configuration
+authors:
+  - SchSeba
+reviewers:
+  - adrianchiris
+  - e0ne
+creation-date: 18-07-2023
+last-updated: 18-07-2023
+---
+
+# Parallel SR-IOV configuration
+
+## Summary
+Allow SR-IOV Network Operator to configure more than one node at the moment.
+
+## Motivation
+SR-IOV Network Operator configures SR-IOV one node at a time and one nic at a same time. That means we’ll need to wait
+hours or even days to configure all NICs  on large cluster deployments. Also moving all draining logic to a centralized
+place which will reduce chances of race conditions and bugs that were encountered before in sriov-network-config-daemon
+with draining.
+
+### Use Cases
+
+### Goals
+* Number of drainable nodes should be 1 by default
+* Number of drainable nodes should be configured by pool
+* Nodes pool should be defined by node selector
+* A node could be included into several pools. In this case pool should be 
+  selected by it’s priority like it’s implemented for SriovNetworkNodePolicy
+* Move all drain-related logic into the centralized place
+
+
+### Non-Goals
+Parallel NICs configuration on the same node is out of scope of this proposal
+
+## Proposal
+Introduce nodes pool drain configuration and controller to meet Goals targets.
+
+
+### Workflow Description
+A new Drain controller will be introduced to manage node drain and cordon procedures. That means we don't need to do
+drain and use `drain lock` in config daemon anymore. The overall drain process will be covered by the following states:
+
+```golang
+NodeDrainAnnotation             = "sriovnetwork.openshift.io/state"
+NodeStateDrainAnnotation        = "sriovnetwork.openshift.io/desired-state"
+NodeStateDrainAnnotationCurrent = "sriovnetwork.openshift.io/current-state"
+DrainIdle                       = "Idle"
+DrainRequired                   = "Drain_Required"
+RebootRequired                  = "Reboot_Required"
+Draining                        = "Draining"
+DrainComplete                   = "DrainComplete"
+```
+
+Drain controller will watch for Node annotation, `sriovnetwork.openshift.io/state` 
+and SriovNetworkNodeState annotation `sriovnetwork.openshift.io/desired-state`
+and write the `sriovnetwork.openshift.io/current-state` annotation in the SriovNetworkNodeState.
+
+Config Daemon will read SriovNetworkNodeState annotation `sriovnetwork.openshift.io/desired-state` and write both
```

Comment:
> sriovnetwork.openshift.io/desired-state

you mean sriovnetwork.openshift.io/current-state ?

in #555 config daemon never reads `sriovnetwork.openshift.io/desired-state`

Reply:
yep sorry about that